### DISCUSSION TITLE
How to configure action states in the Property Inspector

### DISCUSSION THREAD

**Discussion by MarcWeustink** [elgatosf/elgato#78] [Help] [answered]
Hi, in the manifest.json one can define multiple states for a certain action (as a test I defined 8 states). Some questions arise:
- In the property inspector only for the first 2 states, label and the image can be edited. How about the others
- Is it possible to get a notification in the PI that a different state is selected
- Is there a way to prevent SD to cycle through the states when the button is pressed ? I want to control the state through the plugin.

Thanks Marc

**Comment by ChekTek** [comment, marked as answer]
Unfortunately, the Stream Deck application only officially supports 2 states at this time. Adding better support for more has been added as a ticket for the Stream Deck core team to prioritize. 

There is no way to directly get notified when the state changes in the property inspector, however you could use the `sendToPropertyInspector` event whenever you change the state in your plugin to let the property inspector know that the state has changed. https://docs.elgato.com/sdk/plugins/events-sent#sendtopropertyinspector

You can prevent Stream Deck from cycling states automatically by using the `DisableAutomaticStates` flag in your `manifest.json` file. https://docs.elgato.com/sdk/plugins/manifest

**Reply by MarcWeustink** [reply]
Thanks for your answer. 
I indeed might have read somewhere that only 2 states were possible. That explains why I, when I started to integrate SD to our application, I manage myself the states and images belonging to them (depending in the configuration, in what context a button is used, it can have multiple states). At at later stage, when reading about states in the manifest documentation, there isn't a note about a limitation, so I tried that. Since if all the states could be configured and handled by SD, I don't have to send the state images from the plugin to the SD each time a state changes. 
BTW when editing of multiple states is made available, it would be nice
- to have a notification in the PI of the current state being edited so the PI can show additional properties for that state.
- to have control which state is edited (to set the current state, to prevent a certain state to be edited)
